common/lib/dependabot/file_updaters/vendor_updater.rb

```diff
@@ -1,78 +1,32 @@
 # frozen_string_literal: true
 
 require "dependabot/dependency_file"
-
+require "dependabot/file_updaters/artifact_updater"
+
+# This class is a specialisation of the ArtifactUpdater which should be used
+# for vendored files so any DependencyFile objects it creates are properly
+# flagged.
+#
+# This flagging ensures that the Updater will handle them correctly when
+# compiling grouped updates.
 module Dependabot
   module FileUpdaters
-    class VendorUpdater
+    class VendorUpdater < ArtifactUpdater
```

Comment:
I considered an alternative approach of adding a new argument to `VendorUpdater` rather than creating a 'generic' `ArtifactUpdater`, something like 

```
def initialise(repo_contents_path:, vendor_dir:, create_vendored_files: true)
```

Although this would work had result in less churn to the API, I felt this didn't make the intent that vendor files need some special handling clear enough and it would extremely easy for a subtle bug if we were to develop a new use for this utility class similar to the calls in the NPM `FileUpdater`.

The split of 'utility class to grab any artefact outside the manifest files' and 'specific class to manage vendored dependencies' seemed like a safer, more intentional interface.